jan2025/helpers.js

```diff
@@ -0,0 +1,369 @@
+// Follows details outlined in gsheet here https://docs.google.com/spreadsheets/d/1Vtxp5BLweDPDooQoNhgOCYjCIBPRYIcOuyArGJRqOkI/edit?gid=0#gid=0
+const migrateFlowData = (flowData) => {
+  const timestamp = `[${new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '')}]`;
+
+  let newFlowData = flowData;
+  let logs = "";
+
+  Object.entries(flowData).forEach(([nodeId, nodeData]) => {
+    // Flags
+    if (nodeData?.["type"] === 200) {
+      // Question or checklists with `flags` array
+      if (nodeData?.["data"]?.["flags"]?.length > 0) {
+        newFlowData[nodeId]["data"]["flags"] = nodeData["data"]["flags"].map(current =>
+          flagChanges[current]
+        );
+        logs += `${timestamp} Updated Question or Checklist flags (node ${nodeId}); `;
+      }
+
+      // Filter options with flag as `val`
+      const current = nodeData?.["data"]?.["val"];
+      if (current && Object.keys(flagChanges).includes(current)) {
+        const proposed = flagChanges[current];
+        newFlowData[nodeId]["data"]["val"] = proposed;
+
+        // Some flags also get updated `text`
+        if (proposed.endsWith(".consentNeeded")) {
+          newFlowData[nodeId]["data"]["text"] = "Consent needed";
+        }
+        if (proposed === "flag.wtt.notice") {
+          newFlowData[nodeId]["data"]["text"] = "Notice";
+        }
+        logs += `${timestamp} Updated Filter option flags (node ${nodeId}); `;
+      }
+    }
+
+    // Filter flagset category change
+    if (nodeData?.["type"] === 500 && nodeData?.["data"]?.["category"] === "Listed building consent") {
+      newFlowData[nodeId]["data"]["category"] = "Works to listed buildings";
+      logs += `${timestamp} Updated Filter category (node ${nodeId}); `;
+    }
+
+    // Article 4s (options and selectable planning constraints)
+    if (nodeData?.["type"] === 200 && nodeData?.["data"]?.["val"]?.startsWith("article4")) {
+      newFlowData[nodeId]["data"]["val"] = nodeData["data"]["val"].replace("article4", "articleFour");
+      logs += `${timestamp} Updated article4 option val (node ${nodeId}); `;
+    }
+
+    if (
+      nodeData?.["type"] === 11 &&
+      (nodeData?.["data"]?.["dataValues"]?.includes("article4") || nodeData?.["data"]?.["dataValues"]?.includes("article4.caz"))
+    ) {
+      newFlowData[nodeId]["data"]["dataValues"] = nodeData["data"]["dataValues"].map(
+        current => current.startsWith("article4") ? current.replace("article4", "articleFour") : current
```

Comment:
```suggestion
        current => current.replace("article4", "articleFour")
```

`replace()` will return the original string untouched if it's not able to update it.

Reply:
Updated !